library/core/src/ptr/non_null.rs

```diff
@@ -1803,4 +1813,41 @@ mod verify {
         let maybe_null_ptr =  if kani::any() { xptr as *mut i32 } else { null_mut() };
         let _ = NonNull::new(maybe_null_ptr);
     }
+
+    #[kani::proof_for_contract(NonNull::as_ptr)]
+    pub fn non_null_check_as_ptr() {
```

Comment:
Since the API under verification (`NonNull::as_ptr`) has no requirements on the data, it is better to use a less-restrictive harness that passes any pointer value, e.g.
```rust
let non_null_ptr: *mut i32 = kani::any::<usize>() as *mut i32;
let ptr = NonNull::new(non_null_ptr).unwrap();
...
```
This guarantees that the API works for all possible pointer values.

Reply:
Ok, thanks I have used it